templates/repo/view_file.tmpl

```diff
@@ -39,6 +39,12 @@
 			{{end}}
 			{{if not .ReadmeInList}}
 				<div class="ui buttons mr-2">
+					{{if and .IsMarkdownBlob .IsMarkup}}
+						<a class="ui mini basic button" href="{{.RepoLink}}/src/{{.BranchNameSubURL}}/{{PathEscapeSegments .TreePath}}?plain=1">{{.i18n.Tr "repo.file_view_source_blob"}}</a>
+					{{end}}
+					{{if and .IsMarkdownBlob (not .IsMarkup)}}
+						<a class="ui mini basic button" href="{{.RepoLink}}/src/{{.BranchNameSubURL}}/{{PathEscapeSegments .TreePath}}">{{.i18n.Tr "repo.file_view_rendered_blob"}}</a>
+					{{end}}
```

Comment:
`{{if and .IsMarkdownBlob (not .IsMarkup)}}` seems not easy to understand ....

And after grepping the codebase, this `IsMarkup` is only used inside this function and this template. I think it can be safely renamed to a better one.

Maybe it could be:

```
{{if .IsMarkupType}}
    {{if .IsMarkupRendering}}
         ...
    {{else}}
         ...
    {{end}}
{{end}}
```